File: golang/338_counting_bits.go
package main

import "fmt"

func countBits(num int) []int {
	var ans []int
	for i := 0; i < num+1; i++ {
		temp := i
		cnt := 0
		for temp > 0 {
			if temp&1 == 1 {
				cnt++
			}
			temp >>= 1
		}
		ans = append(ans, cnt)
	}
	return ans
}

func main() {
	var inp int
	fmt.Scanf("%d", &inp)
	fmt.Println(countBits(inp))
}

File: python/338_counting_bits.py
from typing import List

class Solution:
    def countBits(self, num: int) -> List[int]:
        ans = []
        for i in range(0, num + 1):
            temp = i
            cnt = 0
            while temp:
                if temp&1:
                    cnt += 1
                temp >>= 1
            ans.append(cnt)
        return ans

sol = Solution()
inp = int(input())
print(sol.countBits(inp))